### PR TITLE
Updated User.modified to return password change timestamp.

### PR DESCRIPTION
**Pull request by AntonCroner**

### CODE REVIEW

File: Context/Users/Storage/Context/Users/User.ts [hunk 1]
```diff
@@ -60,33 +67,39 @@ export namespace User {
...
+				twoFactor: patch.twoFactor
+					? {
+							key: patch.twoFactor?.key,
+							recoveryCodes: await twoFactor.hash(patch.twoFactor?.recoveryCodes, context.secret),
+					  }
+					: source.twoFactor,
+				modified: {
+					other: isoly.DateTime.now(),
+					password:
+						patch.password || typeof source.modified != "object" ? isoly.DateTime.now() : source.modified.password,
```
Comment (SiFalorSum) [inline Context/Users/Storage/Context/Users/User.ts:100]: I think you need to separate these logical expressions and set `password` to `source.modified` if patch.password is undefined and modified is a string. Something like
```
patch.password ? isoly.DateTime.now() : typeof source.modified == "string" ? source.modified : source.modified.password
```

**Review by SiFalorSum — CHANGES_REQUESTED** [review]
See comment

**Review by EliasEriksson — COMMENTED** [review]
Looks good for me

File: Context/Users/Storage/Context/Users/index.ts [hunk 2]
```diff
@@ -11,14 +11,20 @@ import { User } from "./User"
...
 		private readonly context: { application: string; secret: string }
 	) {
 		this.get = async () => (this.cache ??= await storage.object.get("data"))
 		this.set = async user =>
-			(this.cache = await storage.object.set("data", { ...user, modified: isoly.DateTime.now() }))
+			(this.cache = await storage.object.set("data", {
+				...user,
+				modified: {
+					other: isoly.DateTime.now(),
+					password: typeof user.modified != "object" ? isoly.DateTime.now() : user.modified.password,
```
Comment (SiFalorSum) [inline Context/Users/Storage/Context/Users/index.ts:25]: Probably a similar change needed here as well

**Review by SiFalorSum — APPROVED** [review]

**Pull request merged**